internal/nmc/module_nmc_controller.go

```diff
@@ -0,0 +1,139 @@
+package nmc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/api"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+//+kubebuilder:rbac:groups="core",resources=nodes,verbs=get;watch;patch
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=nodemodulesconfigs,verbs=get;list;watch;update;patch;create
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=nodemodulesconfigs/status,verbs=get;update;patch
+
+type ModuleNMCHandler interface {
+	handleModuleNMC(ctx context.Context, mld *api.ModuleLoaderData, nodes []v1.Node) error
+}
+
+type moduleNMCHandler struct {
+	reconHelper moduleNMCHandlerHelperAPI
+}
+
+func NewModuleNMCHandler(client client.Client,
+	nmcHelper Helper) ModuleNMCHandler {
+	reconHelper := newModuleNMCHandlerHelper(client, nmcHelper)
+	return &moduleNMCHandler{
+		reconHelper: reconHelper,
+	}
+}
+
+// func (mnr *ModuleNMCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (mnh *moduleNMCHandler) handleModuleNMC(ctx context.Context, mld *api.ModuleLoaderData, nodes []v1.Node) error {
+	var sumErr error
+	for _, node := range nodes {
+		nodeKernelVersion := strings.TrimSuffix(node.Status.NodeInfo.KernelVersion, "+")
+
+		shouldBeOnNode, err := mnh.reconHelper.shouldModuleRunOnNode(node, mld, nodeKernelVersion)
```

Comment:
```suggestion
		shouldBeOnNode, err := mnh.reconHelper.shouldModuleRunOnNode(node, mld)
```

We already have this info in `node`

Reply:
we do, but we have already extracted that data above, why not just use it, instead of extracting it again?

Reply:
Because we are sending the same data twice which is just redundant IMO and may lead to inconsistency in future code changes. What is the cost of extracting it again? This is just a golang struct. It is not like we are re-generating a new API call to k8s.